### PR TITLE
Clean out most of mathoms.c

### PR DESCRIPTION
**Pull request by khwilliamson**
#22691 enabled macros to easily mimic functions.  This commit takes advantage of that to remove most of mathoms.c.  Macros use fewer resources than stub function calls

* I don't know if this set of changes requires a perldelta entry

### CODE REVIEW

File: av.h [hunk 1]
```diff
@@ -236,7 +236,12 @@ to fit one element without extending:
 
 */
 
-#define newAV()	MUTABLE_AV(newSV_type(SVt_PVAV))
+/* sv_upgrade does AvREAL_only():
```
Comment (mauke) [inline av.h]: What does this comment refer to?

Comment (khwilliamson) [reply av.h]: I just moved the existing comment from mathoms.  It was added by

> commit ac572bf4fde55cf563ae338dfe4e12b22ebc6d80
 Author: Nicholas Clark <nick@ccl4.org>
 Date:   Thu Dec 20 19:49:50 2007 +0000
       Perl_newAV() can become a mathom by making newAV() a wrapper around
     newSV_type() and tweaking Perl_sv_upgrade().
          p4raw-id: //depot/perl@32675

Comment (mauke) [reply av.h]: I see. That comment stopped making sense after b9f83d2f7c, when the `sv_upgrade` calls were removed.

Comment (khwilliamson) [reply av.h]: I removed the entire comment

**Pull request merged**